From 63088e9fcc5c9bf3b45a18d12fb66d0345fc3e53 Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Fri, 13 Aug 2021 11:40:56 +0200
Subject: [PATCH 1/2] Do not close randombytes when using libsodium

randombytes_close is not threadsafe and calling it while still in use by a Context can cause a crash.

For implementations using /dev/[u]random, this can leave up to one leftover FD per process.

The libsodium docs suggest that this function rarely needs to be called explicitly.
---
 src/random.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/random.cpp b/src/random.cpp
index 17c3537df3..12dead87ba 100644
--- a/src/random.cpp
+++ b/src/random.cpp
@@ -151,8 +151,6 @@ static void manage_random (bool init_)
     if (init_) {
         int rc = sodium_init ();
         zmq_assert (rc != -1);
-    } else {
-        randombytes_close ();
     }
 #else
     LIBZMQ_UNUSED (init_);
